### PR TITLE
[Implementation][EFIP-11] Async Admin Task Execution for Validator Management

### PR DESCRIPTION
**Pull request by seongyun-ko**
- `executeTasks` now enqueues the tasks for admin tasks so that they can be processed async instead of sync processing them in a single txn
- `_enqueueAdminTask_ApproveValidators` to split the tasks and queue them
- `AdminTask_ApproveValidators` to process the pending admin tasks in LIFO manner

Any feedback, extension welcome

### CODE REVIEW

File: src/EtherFiAdmin.sol [hunk 1]
```diff
@@ -164,6 +170,31 @@ contract EtherFiAdmin is Initializable, OwnableUpgradeable, UUPSUpgradeable {
...
+
+        require(etherFiOracle.isConsensusReached(_reportHash), "EtherFiAdmin: report didn't reach consensus");
+        require(task.operationHash == keccak256(abi.encode(_reportHash, _validatorsToApprove, _pubKey, _signature)), "EtherFiAdmin: invalid task");
+
+        liquidityPool.batchApproveRegistration(_validatorsToApprove, _pubKey, _signature);
+    }
+
+    // FIFO is better?
+    // should we support random-access so that the admin can process the pending transactions in any order?
+    function _dequeuePendingTask(string memory _operation) internal returns (AdminTask memory) {
```
Comment (vvalecha519) [inline src/EtherFiAdmin.sol:185]: maybe this should be an external function with onlyAdmin modifier. What happens if admin wants to unpublished? How do we clear the queue?

Comment (seongyun-ko) [reply src/EtherFiAdmin.sol:185]: good point. we need to add a function to handle such cases

File: src/EtherFiAdmin.sol [hunk 2]
```diff
@@ -153,7 +159,7 @@ contract EtherFiAdmin is Initializable, OwnableUpgradeable, UUPSUpgradeable {
         numValidatorsToSpinUp = _report.numValidatorsToSpinUp;
 
         _handleAccruedRewards(_report);
-        _handleValidators(_report, _pubKey, _signature);
+        _handleValidators(reportHash, _report, _pubKey, _signature);
```
Comment (vvalecha519) [inline src/EtherFiAdmin.sol:162]: should we remove _pubKey in _handleValidators waste of massive argument. It does nothing but get hashed in the _enqueueAdminTask_ApproveValidators. The benefit is that during the same pubkeys are provided as were passed during the executeTask, but if thats wrong then this is wrong as well.

**Review by vvalecha519 — COMMENTED** [review]
Should executeTask revert if the approving validator queue is empty? Otherwise lastHandledReportRefSlot is wrong because the validators are not all proccessed but we claim the reported has been handled

File: src/EtherFiAdmin.sol [hunk 3]
```diff
@@ -189,25 +220,48 @@ contract EtherFiAdmin is Initializable, OwnableUpgradeable, UUPSUpgradeable {
...
         if (_report.validatorsToApprove.length > 0) {
-            liquidityPool.batchApproveRegistration(_report.validatorsToApprove, _pubKey, _signature);
+            _enqueueAdminTask_ApproveValidators(_reportHash, _report.validatorsToApprove,  _pubKey, _signature);
+            // liquidityPool.batchApproveRegistration(_report.validatorsToApprove, _pubKey, _signature);
         }
 
         // liquidityPoolValidatorsToExit
         if (_report.liquidityPoolValidatorsToExit.length > 0) {
-            liquidityPool.sendExitRequests(_report.liquidityPoolValidatorsToExit);
+            // liquidityPool.sendExitRequests(_report.liquidityPoolValidatorsToExit);
```
Comment (vvalecha519) [inline src/EtherFiAdmin.sol:254]: why did we get rid of this?

**Comment by vvalecha519** [comment]
updated setBatchSize to setValidatorTaskBatchSize as you recommended

**Review by vvalecha519 — APPROVED** [review]

**Pull request merged**